CorsixTH/Src/th_gfx.cpp

```diff
@@ -1466,23 +1466,64 @@ void animation::set_animation_kind(animation_kind anim_kind) {
 }
 
 namespace {
+//! Storage of one or two sounds. If two sounds are stored, both are selected in 50% of the cases.
+struct sound_pair {
+  sound_pair(int16_t sound): soundA(sound), soundB(-1) { }
+  sound_pair(int16_t soundA, int16_t soundB): soundA(soundA), soundB(soundB) { }
+
+  int16_t get_sound() const {
+    if (soundB < 0) {
+      return soundA;
+    } else {
+      int value = rand();
+      int counter = 0;
+      for (int i = 0; i < 7; i++) {
+        counter = counter + (value & 1);
+        value >>= 1;
+      }
+      return (counter < 4) ? soundA : soundB;
+    }
+  }
+
+private:
+  const int16_t soundA; //!< First available sound.
+  const int16_t soundB; //!< If non-negative, the second available sound.
+};
+
+typedef std::map<size_t, sound_pair> sound_replacement_map;
+
 // Map of frame numbers to sounds to play.
-const std::map<size_t, int> frame_sound_replacements{
+const sound_replacement_map frame_sound_replacements {
     // Female flying to heaven (anim 3220)
-    {6987, 123},
+    {6987, sound_pair(123)},
+
     // Using Computer (anim 2098)
-    {4213, 35},
-    {4215, 35},
-    {4224, 35},
-    {4230, 35},
+    {4213, sound_pair(35)},
```

Comment:
I'm not convinced `sound_pair` is a necessary call where only one sound is provided? But I also don't know C++ well so maybe it makes more sense to always run this function

Reply:
The `{a, b}` pair are a key/value pair of the `std::map`, which is similar to key/value pair in a lua table except you write an `=` between the values of the pair then.
The key/value pair used to be `size_t, int` (2 integer numbers). I changed it to `size_t, sound_pair` (an integer + class instance). For this reason the second parameter must construct an object.
Lines 1471 and 1472 give 2 options for that. One with 1 sound value, and one with two sound values for the pair.